rewriter/src/main/java/com/facebook/coresql/rewriter/OrderByRewriter.java

```diff
@@ -73,23 +66,26 @@ public void visit(OrderByClause node, Void data)
 
     private static class OrderByPatternMatcher
             extends SqlParserDefaultVisitor
-            implements PatternMatcher<Set<AstNode>>
     {
-        private final Set<AstNode> matchedNodes;
+        private final AstNode root;
+        private final ImmutableSet.Builder<AstNode> builder = ImmutableSet.builder();
         private int depth;
         private static final int MINIMUM_SUBQUERY_DEPTH = 2; // Past this depth, all queries we encounter are subqueries
 
-        public OrderByPatternMatcher()
+        public OrderByPatternMatcher(AstNode root)
         {
-            this.matchedNodes = new HashSet<>();
+            this.root = requireNonNull(root, "AST passed to rewriter was null");
         }
 
-        @Override
-        public Set<AstNode> matchPattern(AstNode node)
+        public static Set<AstNode> getMatchedNodes(AstNode root)
```

Comment:
Why do we need this public static method? Why can't the rewriter just call `new OrderByPatternMatcher(root).matchPattern();`

Reply:
Superfluous; just removed it.

Done